packages/backend-proxy-middleware/src/base/proxy.ts

```diff
@@ -0,0 +1,282 @@
+
+    if (config.client) {
+        const sapClient = 'sap-client=' + config.client;
+        if (targetPath.match(/sap-client=[0-9]{3}/)) {
+            targetPath = targetPath.replace(/sap-client=[0-9]{3}/, sapClient);
+        } else {
+            targetPath = targetPath.indexOf('?') !== -1 ? targetPath + '&' + sapClient : targetPath + '?' + sapClient;
+        }
+    }
+
+    if (bsp) {
+        const regex = new RegExp('(' + bsp + '/manifest\\.appdescr\\b)');
+        // returns manifest.json if manifest.appdescr is requested
+        if (targetPath.match(regex)) {
+            targetPath = '/manifest.json';
+        }
+    }
+
+    if (config.pathPrefix) {
+        log.info(`Rewrite path ${path} > ${targetPath}`);
+    } else {
+        log.info(targetPath);
+    }
+    return targetPath;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const proxyErrorHandler = (err: any, req: Request, _res: ServerResponse): void => {
+    switch (err && err.code) {
+        case 'UNABLE_TO_GET_ISSUER_CERT_LOCALLY': {
+            const errMsg = i18n.t('error.sslProxy');
+            const error = new Error(errMsg);
+            if (typeof req.next === 'function') {
+                req.next(error);
+            } else {
+                throw error;
+            }
+            break;
+        }
+        case undefined:
+            break;
+        default: {
+            if (typeof req.next === 'function') {
+                req.next(err);
+            } else {
+                throw new Error(err);
+            }
+        }
+    }
+};
+
+/**
+ * Initialize i18next with the translations for this module.
+ */
+async function initI18n(): Promise<void> {
+    const ns = 'backend-proxy-middleware';
+    await i18n.init({
+        resources: {
+            en: {
+                [ns]: translations
+            }
+        },
+        lng: 'en',
+        fallbackLng: 'en',
+        defaultNS: ns,
+        ns: [ns]
+    });
+}
+
+/**
+ * Enhance the proxy options and backend configurations for the usage of destinations in SAP Business Application Studio.
+ *
+ * @param proxyOptions reference to a proxy options object that the function will enhance
+ * @param backend reference to the backend configuration that the the function may enhance
+ * @returns true if the destination requires an authentication prompt in the CLI
+ */
+export async function enhanceConfigsForDestination(
+    proxyOptions: Options,
+    backend: DestinationBackendConfig
+): Promise<boolean> {
+    let authNeeded = true;
+    if (backend.destinationInstance) {
+        const url = new URL(getDestinationUrlForAppStudio(backend.destination));
+        url.username = await getUserForDestinationService(backend.destinationInstance);
+        proxyOptions.target = url.href.replace(/\/$/, '');
+        proxyOptions.headers!['bas-destination-instance-cred'] = url.username;
```

Comment:
The credentials have the wrong encoding it needs to be 
```Buffer.from(`${encodeURIComponent(clientId)}:${encodeURIComponent(clientSecret)}`).toString('base64');```. But `getUserForDestinationService` returns only ```Buffer.from(`${clientId}:${clientSecret}`).toString('base64');```.

Reply:
Done